sdk/tests/conformance/glsl/bugs/character-set.html

```diff
@@ -1,138 +1,178 @@
-        return;
-    }
-
-    gl.shaderSource(shader, shaderSource);
-    wtu.glErrorShouldBe(gl, expected, msg);
-}
-
-function runTest() {
-
-    testShaderSource('fs-invalid-0', gl.INVALID_VALUE, 'Unallowed characters detected');
-    testShaderSource('fs-invalid-1', gl.INVALID_VALUE, 'Unallowed characters detected');
-    testShaderSource('fs-invalid-2', gl.INVALID_VALUE, 'Non ASCII characters detected that would be cleared by preprocessing step is still invalid');
-
-    GLSLConformanceTester.runTests([
-        {
-            fShaderId: 'fs-comment-valid',
-            fShaderSuccess: true,
-            linkSuccess: true,
-            passMsg: 'Unallowed characters and UTF-8 characters in comment are valid'
-        },
-        {
-            fShaderId: 'fs-cleared-by-preprocessing-valid',
-            fShaderSuccess: true,
-            linkSuccess: true,
-            passMsg: 'Unsupported shader extensions (with invalid ASCII characters) cleared by preprocessing step is valid'
-        },
-    ]);
-}
-
-debug("");
-var successfullyParsed = true;
-
-</script>
-<script src="../../../js/js-test-post.js"></script>
-
-</body>
-</html>
+<!--
+Copyright (c) 2020 The Khronos Group Inc.
+Use of this source code is governed by an MIT-style license that can be
+found in the LICENSE.txt file.
+-->
+
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>Character Set</title>
+<link rel="stylesheet" href="../../../resources/js-test-style.css"/>
+<script src="../../../js/js-test-pre.js"></script>
+<script src="../../../js/webgl-test-utils.js"></script>
+<script src="../../../js/glsl-conformance-test.js"></script>
+</head>
+<body>
+
+<script id="fs-ident" type="x-shader/x-fragment">
+precision mediump float;
+FOO
+void main() {
+  gl_FragColor = vec4(1, 0, 0, 1);
+}
+</script>
+<script id="fs-comment" type="x-shader/x-fragment">
+precision mediump float;
+// FOO
+void main() {
+  gl_FragColor = vec4(1, 0, 0, 1);
+}
+</script>
+<script id="fs-ifdef" type="x-shader/x-fragment">
+precision mediump float;
+#if 0
+FOO
+#endif
+void main() {
+  gl_FragColor = vec4(1, 0, 0, 1);
+}
+</script>
+
+<script id="fs-invalid-1" type="x-shader/x-fragment">
+precision mediump float;
+一些注释
+void main() {
+  gl_FragColor = vec4(1, 0, 0, 1);
+}
+</script>
+
+<script id="fs-invalid-2" type="x-shader/x-fragment">
+precision mediump float;
+#if defined(NOT_DEFINED_FOO)
+一些注释
+#endif
+void main() {
+  gl_FragColor = vec4(1, 0, 0, 1);
+}
+</script>
+
+<script id="fs-comment-valid" type="x-shader/x-fragment">
+precision mediump float;
+// some comment: asdf1234~!@#$%^&*()-=[];',./{}:"?><_+
+// some comment: 一些注释 (╯‵□′)╯︵┻━┻
+// some comment: \\\\\\\\\
+// some comment: \n\\r\n
+void main() {
+  gl_FragColor = vec4(1, 0, 0, 1);
+}
+</script>
+
+<script id="fs-cleared-by-preprocessing-valid" type="x-shader/x-fragment">
+precision mediump float;
+#if defined(GL_GOOGLE_cpp_style_line_directive)
+#line 9 "foo.txt"
```

Comment:
It looks like this particular case - the #line directive containing quotes - was lost in the test rewrite. Could this be added back in? (I realize it might be redundant with the `#if 0 .. "  #endif` test.)

Reply:
Resolved in the latest update.